packages/core/src/actions/smartAccount/buildUserOperation.ts

```diff
@@ -48,7 +48,7 @@ export async function buildUserOperation<
 
   const signature = account.getDummySignature();
 
-  const nonce = account.getNonce(overrides?.nonceKey);
+  const nonce = toHex(await account.getNonce(overrides?.nonceKey));
```

Comment:
you don't need to do this because during the build uo pipeline run, the struct remains as the Deferrable type and after the pipeline run succeeds, before signing/sending the user operation, the deferrable type struct is hex-lified into the uo request type.

 would revert the changes in this file.

Reply:
Thank you for review.
ok i guess the sdk have been updated for this issue, this was raise at version `3.6.1`
I will revert it but is this PR is needed? if not needed anymore we can close it.